skrub/_datetime_encoder.py

```diff
@@ -316,7 +344,19 @@ def transform(self, column):
             extracted = _get_dt_feature(column, feature).rename(f"{name}_{feature}")
             extracted = sbd.to_float32(extracted)
             all_extracted.append(extracted)
-        return sbd.make_dataframe_like(column, all_extracted)
+
+        _new_features = []
+        for _case, t in self._required_transformers.items():
+            if _case == "day":
+                _feat = _get_dt_feature(column, "day")
+                _feat = sbd.rename(_feat, "day")
```

Comment:
why do we remove the column name?

Reply:
sure but we should also retain the name of the original input column. so if the column was `"when"` we would get `"when_day_spline_1"`